Ds 2876

jrwishart opened the pull request:
Adds two new outputs. The Jaccard Coefficient output and the Correlation output. These outputs are shown from an importance analysis perspective and use the ImportanceTable output. The coefficients are displayed and given a relative importance score as well. 

Requires checking the `flipStatistics@DS-2876` branch and `flipFormat@DS-2876` branch before merging.

jrwishart committed:
DS-2876 Sanitize input for Jaccard output [revdep skip]

Added functions to check outcome and predictor variables are binary when
user has selected Jaccard Coefficient output. If not all variables are
binary then an error is thrown with an informative message. Unit tests
added.

jrwishart committed:
DS-2876 Add Jaccard Coefficients to output [revdep skip]

Added Jaccard Coefficient as possible output. Added code to the
relativeimportance section as Jaccard coefficient output is deemed part
of importance analysis due to the relative importance being computed.
Code is streamlined to fit into the importance analysis outputs
requiring slightly refactoring the importance analysis functions. Simple
tests added for input validation. More tests for Jaccard Coefficients to
appear in future commits.

jrwishart committed:
DS-2876 Point to flipFormat branch

jrwishart committed:
DS-2876 Implement the Correlation output and add tests

Correlation output method implemented leveraging the
flipStatistics::CorrelationsWithSignificance function. Unit tests added
for both Jaccard Coefficients and Correlation. Also minor version bump
for this new feature.

jrwishart committed:
DS-2876 Point to flipStatistics@DS-2876 in travis

Standard errors are required for the correlation significance tests are
they are passed into the Interaction Crosstab tests to integrate into
the code base more easily.

jrwishart committed:
DS-2876 Remove old print code

Code has been streamlined to use the same print call as Shapley and RIA.
Added element to specify the importance type and and removed old custom
print code lines for Jaccard and Correlation

jrwishart committed:
DS-2876 Add explicit version deps in DESC file [ci skip]

jrwishart committed:
DS-2876 Use absolute correlataion in relative importance [revdep skip]

For computing the relative importance in the new Correlation output option
the absolute correlation is used to ensure the relative importance sums to
100 and is easily interpretable.

jrwishart committed:
DS-2876 Point to flipStatistics and Format in DESC [revdep skip]

Use the Remotes in DESC to install right version of dependencies
instead of the environment variable in .travis.yml

mwmclean commented on R/data.R:
```diff
@@ -5,3 +5,77 @@ outcomeVariableFromModel <- function(Regression.object)
+    if (any(is.na(x)))
+        x <- x[!is.na(x)]
+    unique.values <- unique(x)
+    all(c(0, 1) %in% unique.values && length(unique.values) == 2)
```
the last `)` should be before the `&&`. the second condition always has length 1

jrwishart committed:
DS-2876 Update helper function and update docs [revdep skip]

Add documentation for new output options in Regression and update logic
in helper function `checkVariableIsBinary`

mwmclean commented on R/relativeimportance.R:
```diff
@@ -243,3 +233,217 @@ extractRegressionInfo <- function(formula, data, weights, type, signs,
+    y <- jaccard.coef.output$outcome.variable
+    X <- jaccard.coef.output$predictor.variables
+    jaccard.coefs <- jaccard.coef.output$coefficients
+    names(jaccard.coefs) <- variable.names
```
already done inside `computeJaccardCoefficients`

jrwishart replied:
Removed redundant line in https://github.com/Displayr/flipRegression/pull/22/commits/3451e7f88bae91b96f7a707632a57e746e5ef83a

jrwishart committed:
DS-2876 Code review update: Remove redundant line of code [revdep skip]

jrwishart committed:
DS-2876 Update DESCRIPTION to match flipFormat [revdep skip]

Minor version bump to flipFormat and updating minimum version of flipFormat required for flipRegression

jrwishart committed:
Merge branch 'DS-2876' of https://github.com/Displayr/flipRegression into DS-2876
